src/lib.rs

```diff
@@ -509,8 +511,12 @@ fn readline_edit<H: Helper>(
                 s.edit_overwrite_char(c)?;
             }
             Cmd::EndOfFile => {
+                if s.has_hint() || !s.is_default_prompt() {
```

Comment:
Only if line is empty or in vi mode ?
That is when the action is not `edit_delete`.

Reply:
Fixed. But I'm not sure that `if !input_state.is_emacs_mode() && !s.line.is_empty()` makes sense. `Ctrl+D` should be "scroll downwards" in vim, not "submit".